app/models/reservation.rb

```diff
@@ -0,0 +1,26 @@
+class Reservation < ApplicationRecord
+
+  belongs_to :branch
+  belongs_to :customer
+  has_many :line_items, dependent: :destroy, after_add: :update_total
+
+  validates :booking_schedule, presence: true
+
+  before_create :book_tables
```

Comment:
why before_create and not any other callback?

Reply:
We can to use before_callbacks here, so that we can stop reservation creation if we don't have vacant tables.
before_save will run on create as well update, so `before_create` is used

Reply:
before_validation used now.